### PR TITLE
#655 Prohibit usages of 'CharEncoding'

### PR DESCRIPTION
**Pull request by mkordas**
For #655:
* Use regexp rule to prohibit usages of `CharEncoding`
* Add test case

### CODE REVIEW

**Comment by davvd** [comment]
@mkordas Thanks for the pull request, let me find a reviewer..

**Comment by davvd** [comment]
@pinaf please review,thanks

**Comment by pinaf** [comment]
@mkordas merge conflicts

**Comment by mkordas** [comment]
@pinaf conflict solved

File: qulice-checkstyle/src/test/java/com/qulice/checkstyle/CheckstyleValidatorTest.java [hunk 1]
```diff
@@ -52,7 +52,7 @@
  * @author Yegor Bugayenko (yegor@tpc2.com)
  * @version $Id$
  * @since 0.3
- * @checkstyle MultipleStringLiterals (300 lines)
+ * @checkstyle MultipleStringLiterals (400 lines)
```
Comment (pinaf) [inline qulice-checkstyle/src/test/java/com/qulice/checkstyle/CheckstyleValidatorTest.java]: @mkordas ?

**Comment by pinaf** [comment]
@mkordas 1 question

File: qulice-checkstyle/src/test/java/com/qulice/checkstyle/CheckstyleValidatorTest.java (hunk 1 above)
Comment (mkordas) [inline qulice-checkstyle/src/test/java/com/qulice/checkstyle/CheckstyleValidatorTest.java]: @pinaf I've added some code in the middle of `CheckstyleValidatorTest.java`, so I needed to extend suppression a bit

Comment (pinaf) [inline qulice-checkstyle/src/test/java/com/qulice/checkstyle/CheckstyleValidatorTest.java]: @mkordas the diff doesn't show any changes in this file besides this one.

Comment (mkordas) [inline qulice-checkstyle/src/test/java/com/qulice/checkstyle/CheckstyleValidatorTest.java]: @pinaf right, mistake during rebase, please check now

**Comment by pinaf** [comment]
@rultor merge

File: qulice-checkstyle/src/main/resources/com/qulice/checkstyle/checks.xml [hunk 2]
```diff
@@ -149,6 +149,11 @@
         <property name="message" value="Using class as a lock is a bad practice (use class variable instead)"/>
     </module>
     <module name="RegexpSingleline">
+        <property name="format" value="org\.apache\.commons\.(codec|lang3?)\.CharEncoding"/>
```
Comment (krzyk) [inline qulice-checkstyle/src/main/resources/com/qulice/checkstyle/checks.xml]: @mkordas also `lang` (for 2.x commons)

**Comment by krzyk** [comment]
@mkordas as you can see we have to wait for the amount of open tasks lowers

**Comment by mkordas** [comment]
@krzyk where this limit of tasks is set?

**Comment by krzyk** [comment]
@mkordas @davvd has it, he replied few times that current number of tasks (11) in this project is too big

**Comment by mkordas** [comment]
@krzyk yeah, I've seen. But are such limits described in XDSD policy?

**Comment by krzyk** [comment]
@rultor merge pls

**Comment by davvd** [comment]
@pinaf 10 mins added to @krzyk account (our architect), payment ID is `77216311`, since quality is good here; I just added **20 mins** to your account, many thanks for your contribution (AP-7PM046780V839124K).. 166 hours and 2 mins spent here; review comments (c=5) added as a bonus; +20 added to your rating, at the moment it is: [+6687](http://www.netbout.com/b/35128?open=rating)

**Comment by davvd** [comment]
@rultor please deploy

**Pull request merged**